lib/ramble/ramble/repository.py

```diff
@@ -590,120 +682,125 @@ def repo_for_app(self, spec):
                 tty.debug('Found repo...')
                 return repo
 
-        # If the application isn't in any repo, return the one with
+        # If the object isn't in any repo, return the one with
         # highest precedence.  This is for commands like `ramble edit`
-        # that can operate on applications that don't exist yet.
+        # that can operate on objects that don't exist yet.
         return self.first_repo()
 
     @autospec
     def get(self, spec):
-        """Returns the application associated with the supplied spec."""
-        return self.repo_for_app(spec).get(spec)
+        """Returns the object associated with the supplied spec."""
+        return self.repo_for_obj(spec).get(spec)
 
-    def get_app_class(self, app_name):
-        """Find a class for the spec's application and return the class object."""  # noqa: E501
-        return self.repo_for_app(app_name).get_app_class(app_name)
+    def get_obj_class(self, obj_name):
+        """Find a class for the spec's object and return the class object."""  # noqa: E501
+        return self.repo_for_obj(obj_name).get_obj_class(obj_name)
 
     @autospec
     def dump_provenance(self, spec, path):
         """Dump provenance information for a spec to a particular path.
 
            This dumps the package file and any associated patch files.
-           Raises UnknownApplicationError if not found.
+           Raises UnknownObjectError if not found.
         """
-        return self.repo_for_app(spec).dump_provenance(spec, path)
+        return self.repo_for_obj(spec).dump_provenance(spec, path)
 
-    def dirname_for_application_name(self, app_name):
-        return self.repo_for_app(app_name).dirname_for_application_name(
-            app_name)
+    def dirname_for_object_name(self, obj_name):
+        return self.repo_for_obj(obj_name).dirname_for_object_name(
+            obj_name)
 
-    def filename_for_application_name(self, app_name):
-        return self.repo_for_app(app_name).filename_for_application_name(
-            app_name)
+    def filename_for_object_name(self, obj_name):
+        return self.repo_for_obj(obj_name).filename_for_object_name(
+            obj_name)
 
-    def exists(self, app_name):
-        """Whether application with the give name exists in the path's repos.
+    def exists(self, obj_name):
+        """Whether object with the give name exists in the path's repos.
 
-        Note that virtual applications do not "exist".
+        Note that virtual objects do not "exist".
         """
-        return any(repo.exists(app_name) for repo in self.repos)
+        return any(repo.exists(obj_name) for repo in self.repos)
 
     # TODO: DWJ - Maybe we don't need this? Are we going to have virtual
-    #             applications
-    # def is_virtual(self, app_name, use_index=True):
-    #     """True if the application with this name is virtual,
+    #             objects
```

Comment:
Time to delete this?

Reply:
I want to delete this, except Spack needs this. And I think we're going to have some design meetings in the semi-near future around re-designing this. So, even though it's commented out / dead code right now, I want them to see where it would fit when we get to the redesign.